stock_partner_delivery_window/models/stock_picking.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+from odoo.tools.misc import format_datetime
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+    delivery_date_warning_message = fields.Char(
+        compute="_compute_delivery_date_warning_message"
+    )
+
+    def _planned_delivery_date(self):
+        return self.scheduled_date
+
+    @api.onchange("partner_id", "scheduled_date")
+    def _compute_delivery_date_warning_message(self):
+        self.ensure_one()
+        partner = self.partner_id
+        if not partner:
+            self.delivery_date_warning_message = False
+            return
```

Comment:
And set the default value of the computed field to False at the beginning to not setting it in `if/else` statements afterwards:
```suggestion
            picking.delivery_date_warning_message = False
            if not partner:
                continue
```